app/javascript/components/lesson-preview/components/share-button.jsx

```diff
@@ -0,0 +1,39 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import axios from '../../../src/js/axiosWithCsrf';
+
+const ShareButton = ({ content }) => {
+  const [copied, setCopied] = useState(false);
+
+  useEffect(() => {
```

Comment:
Suggestion: Instead of switching back to "Share" after 4 seconds, lets only switch when/if the content changes again:

```jsx
useEffect(() => {
  if (copied) {
    setCopied(false);
  }
}, [content]);
```